beamer/health/check.py

```diff
@@ -0,0 +1,526 @@
+        "deployment_dir": Path(config["deployment-dir"]),
+        "cache_file_path": Path(config["cache-file-path"]),
+        "notification_system": config["notification-system"],
+        "rpcs": rpcs,
+        "explorers": explorers,
+        "notification": config["notification"],
+        "tokens": config["tokens"],
+    }
+
+
+def get_config() -> MonitorConfig:
+    assert global_config is not None
+    return global_config
+
+
+def main(
+    config_path: Path,
+) -> None:
+    setup_logging(log_level="DEBUG", log_json=False)
+
+    set_config(config_path)
+
+    config = get_config()
+
+    ctx = Context()
+    ctx.agent_address = config["agent_address"]
+    ctx.notification_state = NotificationState(config["cache_file_path"])
+    ctx.notification_state.persist()
+
+    transfers = create_transfers_object(fetch_events())
+
+    cleanup_transfers(transfers)
+
+    analyze_transfers(transfers, ctx)
+
+
+def cleanup_transfers(transfers: TransferMap) -> None:
+    for request_id, values in transfers.copy().items():
+        if values.get("created") is None:
+            del transfers[request_id]
+
+        claims = values.get("claimed")
+        if claims is not None:
+            for events in claims.values():
+                events.sort(key=lambda a: -1 if a.last_challenger == ADDRESS_ZERO else 1)
+
+
+def fetch_events() -> ChainEventMap:
+    deployment_info = beamer.agent.contracts.load_deployment_info(get_config()["deployment_dir"])
+    events = {}
+
+    for chain_id, (rpc, start_block) in get_config()["rpcs"].items():
+        web3 = make_web3(rpc)
+        assert chain_id == ChainId(web3.eth.chain_id)
+
+        info = deployment_info[ChainId(chain_id)]
+        contracts = beamer.agent.contracts.make_contracts(web3, info)
+
+        ef = beamer.agent.events.EventFetcher(
+            web3, (contracts["RequestManager"], contracts["FillManager"]), start_block
+        )
+
+        events[chain_id] = ef.fetch()
+
+    return cast(ChainEventMap, events)
+
+
+def make_web3(rpc: str) -> Web3:
+    web3 = Web3(Web3.HTTPProvider(rpc, request_kwargs=dict(timeout=5)))
+    web3.middleware_onion.inject(geth_poa_middleware, layer=0)
+    return web3
+
+
+def get_agent_liquidity(
+    agent_address: str, rpcs: dict, tokens: TokenMap
+) -> dict[str, dict[ChainId, int]]:
+    contract_abi = load_ERC20_abi()
+    liquidity: dict = defaultdict(dict)
+    agent_address = Web3.toChecksumAddress(agent_address)
+
+    for name, chain_to_token_mapping in tokens.items():
+        for chain_id, token_address in chain_to_token_mapping:
+            contract_address = Web3.toChecksumAddress(token_address)
+            web3 = make_web3(rpcs[int(chain_id)][0])
+            contract = web3.eth.contract(address=contract_address, abi=contract_abi)
+            token_decimals = contract.functions.decimals().call()
+            liquidity[name][chain_id] = (
+                contract.functions.balanceOf(agent_address).call() * 1 * 10**-token_decimals
```

Comment:
Nitpick: you don't need the multiplication by `1`.
I would also split the expression for readability e.g.
```python
            balance = contract.functions.balanceOf(agent_address).call()
            liquidity[name][chain_id] = balance * 10**-token_decimals
```